### ISSUE TITLE
BUG | `useDark` | Cannot use useDark with Storybook

### ISSUE THREAD

**Issue by eric-g-97477** [vueuse/vueuse#4397]
### Describe the bug

I do `const isDark = useDark()`. When I try to access the value from my component inside of Storybook (`npm run storybook`), it will always report false even thought I can see the <html> tag contains `class="dark"`. When I run the app with `npm run dev`, it works as expected. The only reason I can think of is that when running in storybook, the component is operating within an iframe.

Is this a known issue? Is there a workaround? Have I not configured something correctly?

### Reproduction

https://github.com/eric-g-97477-vue/storybook-vueuse

### System Info

```Shell
$ npx envinfo --system --npmPackages '{vue,@vueuse/*}' --binaries --browsers                                                                                                                                                      

  System:
    OS: macOS 14.7
    CPU: (16) arm64 Apple M3 Max
    Memory: 61.78 MB / 64.00 GB
    Shell: 3.7.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 20.17.0 - ~/.nvm/versions/node/v20.17.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v20.17.0/bin/yarn
    npm: 10.8.2 - ~/.nvm/versions/node/v20.17.0/bin/npm
    pnpm: 9.15.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 131.0.6778.140
    Safari: 18.0.1
  npmPackages:
    @vueuse/core: ^12.0.0 => 12.0.0
    @vueuse/nuxt: ^12.0.0 => 12.0.0
    vue: ^3.5.13 => 3.5.13
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by babu-ch** [comment]
If it is the same origin, you should be able to specify the parent window.

```js
const isDark = useDark({
  window:window.parent.window,
  disableTransition: false // For some reason an error occurs, so set it to false for now.
});
```

**Comment by eric-g-97477** [comment]
> If it is the same origin, you should be able to specify the parent window.
> 
> ```js
> const isDark = useDark({
>   window:window.parent.window,
>   disableTransition: false // For some reason an error occurs, so set it to false for now.
> });
> ```

Unfortunately, that did not work. In Button.vue, I did:
```
	const isDark = useDark({
		window: window.parent.window,
		disableTransition: false // For some reason an error occurs, so set it to false for now.
	});
```

I then started storybook ( `npm run storybook` ). I then clicked the button, in both light and dark mode. In both cases, it logged the current `isDark.value` as `false`. This is incorrect.

**Comment by btd** [comment]
I am not sure that `window.parent.window` will work with sandboxed iframe.

**Comment by babu-ch** [comment]
ah. It seems you misunderstood the meaning!
I would like to find out if I have time

**Comment by babu-ch** [comment]
@eric-g-97477 
useDark is not able to follow the theme change of storybook, so it seems that it will be necessary to follow it manually.
Like this
```js
	decorators: [
		withThemeByClassName({
			themes: {
				light: 'light',
				dark: 'dark'
			},
			defaultTheme: 'light'
		}),
		(Story, context) => {
			const [globals] = useGlobals()
			const isDark = useDark();
			nextTick(() => {
				isDark.value = globals.theme === 'dark'
			})
			return Story(context)
		},
	]
```

**Comment by eric-g-97477** [comment]
> ```js
> 	const isDark = useDark();
> 			nextTick(() => {
> 				isDark.value = context.globals.theme === 'dark';
> 			});
> 			return Story(context);
> ```

Yep. That one worked. I wonder what was different about our environments. Why were you able to use useGlobals and I was not...?

It is also interesting that useDark does not monitor the DOM. I would have thought that would be necessary. I will have to look into how it is actually working when I get the chance. Perhaps there would be a way to have useDark work regardless of environment.

In any case, thank you for the workaround.

**Comment by OrbisK** [comment]
> > @eric-g-97477 Since useDark does not monitor the DOM, I think there is a discrepancy with the storybook theme. Oops, there was no error in my environment, but what about this?
> 
> Oh, I did see one more issue with this solution, but it is a minor one. There is an initialization problem. I did:
> 
> ```
> 		(Story, context) => {
> 			const isDark = useDark();
> 			nextTick(() => {
> 				console.log(
> 					'🚀 ~ file: preview.ts:29 ~ nextTick ~ context.globals.theme:',
> 					context.globals.theme
> 				);
> 
> 				isDark.value = context.globals.theme === 'dark';
> 			});
> 			return Story(context);
> 		}
> ```
> 
> When the story is first run (make sure you use a new browser window), what is logged is:
> 
> ```
> 🚀 ~ file: preview.ts:29 ~ nextTick ~ context.globals.theme: 
> ```
> 
> `context.globals.theme` has not been initialized. This code is executed again when the theme changes by pressing the button. After that, it works without issue.

if the function can be async you can `await nextTick()` so you dont return the context before next tick

**Comment by OrbisK** [comment]
this does not seem to be a vueuse issue. ill transfer this into a discussion.
